### PR TITLE
VIDCS-974: React Native- IOS: App crash

### PR DESCRIPTION
**Pull request by jvalli**
`OTScreenCapture` class was updated to replace timer producing crash on start/stop screen share publish.

### Contributing checklist
- [√] Code must follow existing styling conventions
- [√] Added a descriptive commit message
- [x] Sample apps updated if needed

### Solves issue(s)
[VIDCS-974: React Native- IOS: App crash](https://jira.vonage.com/browse/VIDCS-974)

### CODE REVIEW

File: ios/OpenTokReactNative/OTScreenCapture.m [hunk 1]
```diff
@@ -30,117 +35,131 @@ - (instancetype)initWithView:(UIView *)view
...
 
 #pragma mark - Capture lifecycle
 
 /**
- * Allocate capture resources; in this case we're just setting up a timer and 
+ * Allocate capture resources; in this case we're just setting up a timer and
  * block to execute periodically to send video frames.
  */
+
 - (void)initCapture {
```
Comment (goncalocostamendes) [inline ios/OpenTokReactNative/OTScreenCapture.m:144]: being this empty maybe we should remove the funtion description

File: ios/OpenTokReactNative/OTScreenCapture.m [hunk 2]
```diff
@@ -30,117 +35,131 @@ - (instancetype)initWithView:(UIView *)view
...
+        });
+    });
+}
+
+- (void)createCGContextFromPixelBuffer {
+    CGColorSpaceRef rgbColorSpace = CGColorSpaceCreateDeviceRGB();
+    CVPixelBufferLockBaseAddress(_pixelBuffer, 0);
     
-    NSParameterAssert(status == kCVReturnSuccess && _pixelBuffer != NULL);
+    _bitmapContext = NULL;
```
Comment (goncalocostamendes) [inline ios/OpenTokReactNative/OTScreenCapture.m:103]: seems unnecessary this line

**Review by goncalocostamendes — APPROVED** [review]
LGTM! Lef some comments but feel free to disregard

File: ios/OpenTokReactNative/OTScreenCapture.m (hunk 2 above)
Comment (jvalli) [reply ios/OpenTokReactNative/OTScreenCapture.m]: Updated!

File: ios/OpenTokReactNative/OTScreenCapture.m (hunk 1 above)
Comment (jvalli) [reply ios/OpenTokReactNative/OTScreenCapture.m:138]: Updated!

File: ios/OpenTokReactNative/OTScreenCapture.m [hunk 3]
```diff
@@ -30,117 +35,125 @@ - (instancetype)initWithView:(UIView *)view
...
-- (void)checkImageSize:(CGImageRef)image {
-    CGFloat width = CGImageGetWidth(image);
-    CGFloat height = CGImageGetHeight(image);
-    
-    if (_videoFrame.format.imageHeight == height &&
-        _videoFrame.format.imageWidth == width)
-    {
-        // don't rock the boat. if nothing has changed, don't update anything.
+- (CMTime)getTimeStamp {
+    static mach_timebase_info_data_t time_info;
```
Comment (juliobecerragomez) [inline ios/OpenTokReactNative/OTScreenCapture.m:61]: Why do we need a static here? Is the call below particularly slow?

**Review by juliobecerragomez — APPROVED** [review]
Not much to say. I am not familiar with this code, if it fixes the crash... great!

**Pull request merged**